onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -384,27 +301,120 @@
+                params.StockhamParams.ResourceLoopList.back().Sizes = temporarySize;
+                params.StockhamParams.ResourceLoopList.back().Strides = temporaryStrides;
+
+                auto& resource = params.StockhamParams.ResourceLoopList.back().Resource;
+                ORT_THROW_IF_FAILED(context->AllocateTemporaryData(temporaryBufferByteSize, &resource));
+            }
+
+            // If 2 temps, add another
+            if (oscillateBetweenTwoTemporaries)
+            {
+                params.StockhamParams.ResourceLoopList.push_back({});
+                params.StockhamParams.ResourceLoopList.back().Sizes = temporarySize;
+                params.StockhamParams.ResourceLoopList.back().Strides = temporaryStrides;
+
+                auto& resource = params.StockhamParams.ResourceLoopList.back().Resource;
+                ORT_THROW_IF_FAILED(context->AllocateTemporaryData(temporaryBufferByteSize, &resource));
+            }
+
+            // Add output resource
+            params.StockhamParams.ResourceLoopList.push_back({});
+            params.StockhamParams.ResourceLoopList.back().Resource = outputResource;
+            params.StockhamParams.ResourceLoopList.back().Sizes = reshapedOutputSize;
+            params.StockhamParams.ResourceLoopList.back().Strides = reshapedOutputStrides;
+            params.StockhamParams.OutputIndex = static_cast<uint32_t>(params.StockhamParams.ResourceLoopList.size() - 1);
+
+            // Add the temporary after output incase of odd number of passes
+            if (oscillateFirstOutputThenTemporary)
+            {
+                params.StockhamParams.ResourceLoopList.push_back({});
+                params.StockhamParams.ResourceLoopList.back().Sizes = temporarySize;
+                params.StockhamParams.ResourceLoopList.back().Strides = temporaryStrides;
+                
+                auto& resource = params.StockhamParams.ResourceLoopList.back().Resource;
+                ORT_THROW_IF_FAILED(context->AllocateTemporaryData(temporaryBufferByteSize, &resource));
+            }
+
+            // Define the loop range
+            if (writeToOutput) { params.StockhamParams.LoopRange = { 0, 1, params.StockhamParams.NumberOfPasses }; }
+            if (oscillateBetweenTwoTemporaries) { params.StockhamParams.LoopRange = { 1, 2, params.StockhamParams.NumberOfPasses }; }
+            if (oscillateFirstOutputThenTemporary) { params.StockhamParams.LoopRange = { 1, 2, params.StockhamParams.NumberOfPasses + 1 }; }
+            if (oscillateFirstTemporaryThenOutput) { params.StockhamParams.LoopRange = { 1, 2, params.StockhamParams.NumberOfPasses + 1 }; }
         }
+
+        return params;
     }
 
-    void StockhamFFT(
-        ID3D12Resource* inputResource,
-        ID3D12Resource* outputResource,
-        ID3D12GraphicsCommandList* commandList)
+    void StockhamFFT(DFTParameters& dftParams, ID3D12GraphicsCommandList* commandList)
```

Comment:
>& [](http://example.com/codeflow?start=34&length=1)

`const&`? It seems `dftParams` is only read from anyway.

Reply:
done